src/parser.c

```diff
@@ -0,0 +1,168 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   parser.c                                           :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: sqiu <sqiu@student.42vienna.com>           +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/06/23 13:13:28 by sqiu              #+#    #+#             */
+/*   Updated: 2023/07/26 01:25:52 by sqiu             ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+/**
+ * @file parser.c
+ * @brief 
+ */
+
+#include "mod_parser.h"
+
+/**
+ * @brief 			Driver function to parse token list.
+ * 
+ * Creates cmd structures and fills them with the token list.
+ * @param lst 		Pointer to list of tokens.
+ * @param cmd 		Pointer pointer to first node of cmd list.
+ * @return t_err 	ERR_MALLOC, SUCCESS
+ */
+t_err	ft_parser(t_tkn_list *lst, t_cmd **cmd)
+{
+	t_err	err;
+	t_cmd	*new;
+	bool	exe_found;
+	bool	cmd_complete;
+
+	exe_found = 0;
+	cmd_complete = 0;
+	new = NULL;
+	err = ft_init_cmd(&new);
+	if (err != SUCCESS)
+		return (err);
+	while (lst)
+	{
+		err = ft_categorise(&lst, new, &exe_found, &cmd_complete);
+		if (err != SUCCESS)
+			return (err);
+		if (cmd_complete)
+			new = ft_new_cmd(new, cmd, &exe_found, &cmd_complete);
+		if (!new)
+			return (ERR_MALLOC);
+		lst = lst->next;
+	}
+	return (ft_finish_cmd_list(new, cmd));
+}
+
+/**
+ * @brief Adds current cmd to cmd_list and creates a new cmd struct.
+ * 
+ * @param curr 			Current cmd struct.
+ * @param cmd 			List of cmd structs.
+ * @param exe_found 	Bool to indicate if executable has been found.
+ * @param cmd_complete 	Bool to indicate if cmd ist complete.
+ * @return t_cmd* 
+ */
+t_cmd	*ft_new_cmd(t_cmd *curr, t_cmd **cmd, bool *exe_found,
+	bool *cmd_complete)
+{
+	t_cmd	*new;
+	t_err	err;
+
+	new = NULL;
+	curr->args = ft_split(curr->arg_buf, ' ');
```

Comment:
ft_split() doesn't work, see down below. Problem are args which have spaces inbetween

Reply:
How about assigning a different IFS? a greek semicolon for all I care

Reply:
I have to admit I don't think that's very elegant. Though it probably won't matter in the scope of the subject and it's an easy fix, so yeah go for it.
Maybe use a macro so we can change it easily if we encounter a problem with a specific separator?

Reply:
You will see, I will choose a beautiful one ;)